src/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl.py

```diff
@@ -718,6 +720,67 @@ def disable_freeu(self):
         """Disables the FreeU mechanism if enabled."""
         self.unet.disable_freeu()
 
+    def enable_fused_qkv_projections(self, unet: bool = True, vae: bool = True):
+        """
+        Enables fused QKV projections. For self-attention modules, all projection matrices (i.e., query,
+        key, value) are fused. For cross-attention modules, key and value projection matrices are fused.
+
+        <Tip warning={true}>
+
+        This API is 🧪 experimental.
+
+        </Tip>
+
+        Args:
+            unet (`bool`, defaults to `True`): To apply fusion on the UNet.
+            vae (`bool`, defaults to `True`): To apply fusion on the VAE.
+        """
+        from ...models.attention_processor import FusedAttnProcessor2_0
+
+        self.fusing_unet = False
+        self.fusing_vae = False
+
+        if unet:
+            self.fusing_unet = True
+            self.unet.enable_fused_qkv_projections(device=self.device, dtype=self.dtype)
```

Comment:
```suggestion
            self.unet.fuse_qkv_projections(device=self.device, dtype=self.dtype)
```
(nit) think "fuse_..." is a bit more readable than "enable_fused_..."

Why is device and dtype passed?